src/AWSCredentials.jl

```diff
@@ -56,392 +58,460 @@ mutable struct AWSCredentials
-    source_profile = get(ini, profile, "source_profile")
+"""
+    _aws_get_credential_details(profile::AbstractString, ini::Inifile) -> Tuple
+
+Get `AWSCredentials` for the specified `profile` from the `inifile`. If targeting the
+`~/.aws/config` file, with a non-default `profile`, you must specify `config=true` otherwise
+the default credentials will be returned.
 
-    profile = "profile $profile"
-    role_arn = get(ini, profile, "role_arn", role_arn)
-    source_profile = get(ini, profile, "source_profile", source_profile)
+# Arguments
+- `profile::AbstractString`: Specific profile used to get AWSCredentials
+- `ini::Inifile`: Inifile to look into for the `profile` credentials
+"""
+function _aws_get_credential_details(profile::AbstractString, ini::Inifile)
+    access_key = _get_ini_value(ini, profile, "aws_access_key_id")
+    secret_key = _get_ini_value(ini, profile, "aws_secret_access_key")
+    token = _get_ini_value(ini, profile, "aws_session_token"; default_value="")
 
-    (source_profile, role_arn)
+    return (access_key, secret_key, token)
 end
 
-function aws_get_credential_details(profile::AbstractString, ini::Inifile, config::Bool)
-    if debug_level > 0
-        filename = config ? dot_aws_config_file() : dot_aws_credentials_file()
-        println("Loading \"$profile\" AWSCredentials from " * filename
-                * "... ")
-    end
 
-    key_id = get(ini, profile, "aws_access_key_id")
-    key = get(ini, profile, "aws_secret_access_key")
-    token = get(ini, profile, "aws_session_token", "")
+"""
+    aws_get_region(profile::AbstractString, ini::Inifile) -> String
 
-    if config
-        profile = "profile $profile"
-        key_id = get(ini, profile, "aws_access_key_id", key_id)
-        key = get(ini, profile, "aws_secret_access_key", key)
-        token = get(ini, profile, "aws_session_token", token)
-    end
+Retrieve the AWS Region for a given profile, returns `us-east-1` as a default.
 
-    (key, key_id, token)
-end
+# Arguments
+- `profile::AbstractString`: Specific profile used to get the region
+- `ini::Inifile`: Inifile to look in for the region
+"""
+function aws_get_region(profile::AbstractString, ini::Inifile)
+    region = get(ENV, "AWS_DEFAULT_REGION", "us-east-1")
+    region = _get_ini_value(ini, profile, "region"; default_value=region)
 
-function aws_get_profile()
-    get(ENV, "AWS_DEFAULT_PROFILE", get(ENV, "AWS_PROFILE", "default"))
+    return region
 end
 
-function aws_get_region(profile::AbstractString, ini::Inifile)
-    region = get(ENV, "AWS_DEFAULT_REGION", "us-east-1")
 
-    region = get(ini, profile, "region", region)
-    region = get(ini, "profile $profile", "region", region)
+"""
+    aws_user_arn(aws::AWSConfig) -> String
+
+Retrieve the user ARN from the `AWSConfig`, if not present query Security Token Services
+(STS).
+
+# Arguments
+- `aws::AWSConfig`: SymbolDict used to retrieve the user arn
+"""
+function aws_user_arn(aws::AWSConfig)
```

Comment:
Split the STS call and field update into a separate function called by this and `aws_account_number`